moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h

```diff
@@ -69,6 +69,18 @@ class JointConstraintSampler : public ConstraintSampler
   JointConstraintSampler(const planning_scene::PlanningSceneConstPtr& scene, const std::string& group_name)
     : ConstraintSampler(scene, group_name)
   {
+    int rng_seed;
+    if (ros::param::get("~constraint_sampler_random_seed", rng_seed))
+    {
+      ROS_WARN_STREAM_NAMED("constraint_samplers",
+                            "Creating random number generator with seed " << std::to_string(rng_seed));
+      random_number_generator_ =
+          std::make_unique<random_numbers::RandomNumberGenerator>(static_cast<uint32_t>(rng_seed));
```

Comment:
Instead of casting here, please use `uint` as the primary data type for _rng_seed_.